webapp/channels/src/components/select_team/components/select_team_item.tsx

```diff
@@ -87,13 +84,15 @@ export default class SelectTeamItem extends React.PureComponent<Props> {
                 >
                     <span className='signup-team-dir__name'>{team.display_name}</span>
                     {!team.allow_open_invite &&
-                        <LocalizedIcon
+                        <i
                             className='fa fa-lock light'
-                            title={{id: t('select_team.private.icon'), defaultMessage: 'Private team'}}
+                            title={this.props.intl.formatMessage({id: 'select_team.private.icon', defaultMessage: 'Private team'})}
                         />}
                     {canJoin && icon}
                 </a>
             </div>
         );
     }
 }
+
+export default injectIntl(SelectTeamItem);
```

Comment:
This is a class component hence we cannot use hooks but the hoc `injectIntl` to access formatMessage function.